src/main/java/com/example/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,91 @@
+import com.example.bookstore.model.ShoppingCart;
+import com.example.bookstore.model.User;
+import com.example.bookstore.repository.CartItemRepository;
+import com.example.bookstore.repository.OrderItemRepository;
+import com.example.bookstore.repository.OrderRepository;
+import com.example.bookstore.repository.ShoppingCartRepository;
+import com.example.bookstore.repository.UserRepository;
+import com.example.bookstore.service.OrderService;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final CartItemRepository cartItemRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+
+    @Transactional
+    @Override
+    public OrderDto create(AddOrderRequestDto requestDto, Long userId) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId)
+                .orElseThrow(() ->
+                        new EntityNotFoundException("Can't find user cart by id"
+                        + userId));
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find user by id" + userId));
+        String shippingAddress = requestDto.getShippingAddress();
+        Order order = new Order();
+        order.setOrderDate(LocalDateTime.now());
+        order.setUser(user);
+        order.setStatus(Order.Status.valueOf(PENDING.name()));
+        order.setShippingAddress(shippingAddress);
+        Set<CartItem> cartItems = shoppingCart.getCartItem();
+        Set<OrderItem> orderItems = new HashSet<>();
+        OrderItem orderItem = new OrderItem();
+        BigDecimal total = BigDecimal.ZERO;
+        orderItem.setOrder(order);
+        for (CartItem cartItem : cartItems) {
+            BigDecimal price = cartItem.getBook().getPrice();
+            orderItem.setQuantity(cartItem.getQuantity());
+            orderItem.setBook(cartItem.getBook());
+            orderItem.setPrice(price);
+            total = total.add(price);
+            orderItems.add(orderItem);
+        }
+        order.setOrderItem(orderItems);
+        order.setTotal(total);
+        orderRepository.save(order);
+        orderItemRepository.saveAll(orderItems);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public List<OrderDto> getAll(Long userId) {
+        List<Order> orders = orderRepository.findAllByUserId(userId);
+        return orders.stream()
+                .map(orderMapper::toDto)
```

Comment:
it's better to use method annotated as `@IterableMapping` inside mapper for mappings of `Iterable` objects